assets/javascripts/discourse/initializers/qa-edits.js

```diff
@@ -53,6 +54,29 @@ function initPlugin(api) {
     },
   });
 
+  api.modifyClass("model:topic", {
+    pluginId,
+
+    @discourseComputed("last_read_post_number", "highest_post_number", "url")
+    lastUnreadUrl(lastReadPostNumber, highestPostNumber) {
```

Comment:
Is there an interface we can introduce in core so that we do not have to override a function in core? One of the goals I wanted to achieve for this plugin is to avoid any overrides.

Reply:
Hmm, I can only think of one not very elegant way to do this. In core, we could add a `customLastUnreadUrl` property, set to nil by default. Then `lastUnreadUrl()` can return that property if/when the plugin sets it. 

It doesn't sound like a better solution than `modifyClass` to me because it would be introducing a new kind of pattern. I personally prefer sticking with `modifyClass`, I hear you on reducing its usage, but the current use case is justified IMO.

Reply:
Ok, let's try with a callback via the plugin API. Needs https://github.com/discourse/discourse/pull/16222